language/move-prover/src/boogie_helpers.rs

```diff
@@ -447,17 +461,25 @@ pub fn boogie_debug_track_abort_via_function(
     pos: &str,
     abort_code: &str,
 ) -> String {
-    format!(
-        "if (true) {{ assume $DebugTrackAbort({}, {}, {}); }}",
+    ensure_trace_info(format!(
+        "assume $DebugTrackAbort({}, {}, {});",
         file_idx, pos, abort_code
-    )
+    ))
 }
 
 /// Construct a statement to debug track an abort using the Boogie attribute approach.
 pub fn boogie_debug_track_abort_via_attrib(file_idx: &str, pos: &str, abort_code: &str) -> String {
-    format!(
+    ensure_trace_info(format!(
         "$trace_abort_temp := {};\n\
-        assume {{:print \"$track_abort({}, {}):\", $trace_abort_temp}} true;",
+        assume {{:print \"$track_abort({},{}):\", $trace_abort_temp}} true;",
         abort_code, file_idx, pos,
-    )
+    ))
+}
+
+/// Wraps a statement such that Boogie does not elimiate it in execution traces. Boogie
+/// does not seem to account for `assume` statements in execution traces, so in order to
+/// let our tracking statements above not be forgotten, this trick ensures that they always
+/// appear in the trace.
+fn ensure_trace_info(s: String) -> String {
+    format!("if (true) {{\n {}\n}}", s.replace("\n", "\n  "))
```

Comment:
You might not care too much about performance but if you did, and all calls to this function are of the form `ensure_trace_info(format!(...))` you could have this function take in `std::fmt::Arguments` and change callers to do `ensure_trace_info!(format_args!(..))` in order to avoid the extra allocation. Of course if you are required to replace `"\n"` with `"\n "` then you may not even be able to do what I've suggested

Reply:
Thanks for the hint, will remember this next time!